utils/update-next.js

```diff
@@ -1,68 +1,160 @@
+/* eslint-disable no-console */
+
+const { dirname, join } = require('path');
 const { execFileSync } = require('child_process');
+const { glob } = require('glob');
+const { pathExists, readJson, writeJson } = require('fs-extra');
 
 function exec(cmd, args, opts) {
-  // eslint-disable-next-line no-console
-  console.log({ input: `${cmd} ${args.join(' ')}` });
+  console.log({ input: `${cmd} ${args.join(' ')}`, cwd: opts?.cwd });
   const output = execFileSync(cmd, args, opts).toString().trim();
   console.log({ output });
   console.log();
   return output;
 }
 
-module.exports = async ({ github, context }) => {
-  const oldVersion = require('../examples/nextjs/package.json').dependencies
-    .next;
+module.exports = async ({ github, context } = {}) => {
   const newVersion = exec('npm', ['view', 'next', 'dist-tags.latest']);
   const branch = `next-${newVersion.replaceAll('.', '-')}`;
 
-  if (oldVersion === newVersion) {
-    // eslint-disable-next-line no-console
+  if (
+    github &&
+    exec('git', ['ls-remote', '--heads', 'origin', branch]).toString().trim()
+  ) {
+    console.log(`Branch ${branch} already exists, skipping update.`);
+    return;
+  }
+
+  let updatedCount = 0;
+
+  // update `examples/nextjs`
+  const oldVersion = require('../examples/nextjs/package.json').dependencies
+    .next;
+  if (github && oldVersion !== newVersion) {
+    updatedCount++;
+    exec('rm', ['-rf', './examples/nextjs']);
+    exec('npx', ['--yes', 'create-next-app@latest', './examples/nextjs']);
+    exec('git', [
+      'config',
+      '--global',
+      'user.email',
+      'infra+release@vercel.com',
+    ]);
+    exec('git', ['config', '--global', 'user.name', 'vercel-release-bot']);
+    exec('git', ['checkout', 'main']);
+    exec('git', ['checkout', '-b', branch]);
+    exec('git', ['add', '-A']);
+    exec('git', ['commit', '-m', branch]);
+    exec('git', ['push', 'origin', branch]);
+  }
+
+  // update tests
+  const testGlobs = [
+    'packages/cli/test/dev/fixtures/*/package.json',
+    'packages/cli/test/fixtures/unit/**/package.json',
+    'packages/fs-detectors/test/fixtures/**/package.json',
+    'packages/next/test/fixtures/*/package.json',
+    'packages/next/test/integration/*/package.json',
+    'packages/static-build/test/fixtures/*/package.json',
+  ];
```

Comment:
I wonder if we could use the new recursive readdir that @Ethan-Arrowood shipped in Node.js core instead of `glob`

Reply:
Not sure when or if thats getting backported so we'd have to be running node 20 to get access to it

Reply:
I removed the dependency on `glob` in favor of a iterative, non-recursive solution. This should be more efficient and faster than a recursive solution.